istioctl/pkg/waypoint/waypoint.go

```diff
@@ -235,6 +242,10 @@ func Cmd(ctx cli.Context) *cobra.Command {
 					fmt.Fprintf(cmd.OutOrStdout(), "Warning: namespace is not enrolled in ambient. Consider running\t"+
 						"`"+"kubectl label namespace %s istio.io/dataplane-mode=ambient"+"`\n", ns)
 				}
+				if hasSidecarInjection {
+					fmt.Fprintf(cmd.OutOrStdout(), "Warning: namespace is enrolled in sidecar mode. It takes precedence over ambient mode. Consider running\t"+
+						"`"+"kubectl label namespace %s istio-injection=disabled"+"`\n", ns)
```

Comment:
I'm not sure in this case we want to prescribe disabling sidecar injection tbh. We're really guessing that trying to add a waypoint means they want ambient for their entire namespace and are fine to lose their sidecars but I'm not convinced that is really a safe assumption to make. Would it potentially make more sense to link users to a more nuanced doc which can explain more scenarios vs having this short inlined prescription?

Reply:
To add to Ian's point, sidecar interoperability is a goal of Ambient. This warning (if I'm reading it correctly) seems to contradict that goal or at least conveys the idea that you can only have sidecar-mode or ambient-mode; not sure we want to tell the user that.

Reply:
If a namespace is labeled with both sidecar injection and dataplane-mode: ambient, how a request will be directed for L7 processing ?

Will it go through a Waypoint Gateway (processing traffic for services) and be directed to an application pod sidecar proxy container and then to the application container ?

```
The Gateway resource is labeled with istio.io/waypoint-for: service, indicating the waypoint can process traffic for services, which is the default.
```

Reply:
It should depend on if the service the traffic is addressed to has a waypoint. If the service uses a waypoint then Istio should honor that even if the backing pod may have a sidecar as well. The output of the waypoint will be HBONE tunneled so a sidecar or ztunnel would need to handle the tunneling before sending the inner/tunneled traffic to the dst.

This is the ideal case. There is work to do to reach this I think but its coming along pretty well and any way it should be our north star IMO.